scripts/run_with_retry

```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 run_with_retry(){
-    [ "$SONIC_BUILD_RETRY_COUNT" -gt 0 ] || SONIC_BUILD_RETRY_COUNT=0
+    (( SONIC_BUILD_RETRY_COUNT > 0 )) || SONIC_BUILD_RETRY_COUNT=0
```

Comment:
@liushilongbuaa IMHO, this is a bad design. Here we are hiding an active build configuration from user, plus some potential issues (e.g., invalid param input). I would recommend adding these params (both `SONIC_BUILD_RETRY_COUNT` and `SONIC_BUILD_RETRY_INTERVAL`) to the `rules/config`.

Another option is:
```bash
SONIC_BUILD_RETRY_COUNT="${SONIC_BUILD_RETRY_COUNT:-0}"
SONIC_BUILD_RETRY_INTERVAL="${SONIC_BUILD_RETRY_INTERVAL:-600}"
```

Please check:
1. `sonic-net/sonic-buildimage/.azure-pipelines/template-variables.yml`
2. `sonic-net/sonic-buildimage/Makefile.work`
3. `sonic-net/sonic-buildimage/slave.mk`
4. `sonic-net/sonic-buildimage/rules/config`

Reply:
If we put it in rules/config, the change in Makefile seems to be heavy. We need to include rules/config and export variables.
It is not a SONiC related config. It is a build related config. We want to improve PR checks build success rate. I think it is not neccessary to put it in rules/config.

Reply:
If input is string. Line 4 will change it to 0.
SONIC_BUILD_RETRY_COUNT="${SONIC_BUILD_RETRY_COUNT:-0}"   this format will keep string variable.

Reply:
https://github.com/sonic-net/sonic-buildimage/pull/12325/files
This change didn't touch slave.mk and Makefile.work

Reply:
> If input is string. Line 4 will change it to 0. SONIC_BUILD_RETRY_COUNT="${SONIC_BUILD_RETRY_COUNT:-0}" this format will keep string input.

Exactly. As i wrote above - we are hiding an issues from user (e.g., expected integer - provided string).
Do we really want to do that?

Reply:
@liushilongbuaa got your point

Reply:
Developer maybe don't need this feature, because it will take long time to retry. But it will be good to PR validation and official build.